internal/remote/sources.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/shared"
+)
+
+// RecentGeneration represents a recent generation of a target in the workspace.
+// The source of this data is our CLI event stream, which is updated every time
+// a target is generated.
+type RecentGeneration struct {
+	CreatedAt            time.Time
+	ID                   string
+	TargetName           string
+	Target               string
+	SourceNamespace      string
+	SourceRevisionDigest string
+	Success              bool
+	Published            bool
+
+	// May not be set
+	GitRepoOrg *string
+	GitRepo    *string
+
+	// gen.yaml
+	GenerateConfig *string
+}
+
+const (
+	// The event stream contains multiple events for the same namespace, so we want to
+	// break execution once we've seen a minimum, arbitrary number of unique namespaces.
+	recentGenerationsToShow int = 5
+)
+
+// GetRecentWorkspaceGenerations returns the most recent generations of targets in a workspace
+// This is based on the CLi event stream, which is updated on every CLI interaction.
+func GetRecentWorkspaceGenerations(ctx context.Context) ([]RecentGeneration, error) {
+	workspaceId, err := core.GetWorkspaceIDFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	speakeasyClient, err := sdk.InitSDK()
+	if err != nil {
+		return nil, err
+	}
+
+	// The event stream is limited to the most recent 250 events
+	res, err := speakeasyClient.Events.SearchWorkspaceEvents(ctx, operations.SearchWorkspaceEventsRequest{
+		WorkspaceID:     &workspaceId,
+		InteractionType: shared.InteractionTypeTargetGenerate.ToPointer(),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res.CliEventBatch) == 0 {
+		return nil, fmt.Errorf("no events found for workspace %s", workspaceId)
+	}
+
+	seenUniqueNamespaces := make(map[string]bool)
+
+	var generations []RecentGeneration
+
+	for _, event := range res.CliEventBatch {
+		// Filter out cli events that aren't generation based, or lack the required
+		// fields.
+		if !isRelevantGenerationEvent(event) {
+			continue
+		}
+
+		if seenUniqueNamespaces[*event.SourceNamespaceName] {
+			continue
+		}
+
+		seenUniqueNamespaces[*event.SourceNamespaceName] = true
+
+		generations = append(generations, RecentGeneration{
+			ID:                   event.ID,
+			CreatedAt:            event.CreatedAt,
+			TargetName:           *event.GenerateTargetName,
+			Target:               *event.GenerateTarget,
+			GitRepoOrg:           event.GitRemoteDefaultOwner,
+			GitRepo:              event.GitRemoteDefaultRepo,
+			SourceNamespace:      *event.SourceNamespaceName,
+			SourceRevisionDigest: *event.SourceRevisionDigest,
+			GenerateConfig:       event.GenerateConfigPreRaw,
+			Success:              event.Success,
+		})
+
+		if len(seenUniqueNamespaces) >= recentGenerationsToShow {
+			break
+		}
+	}
+
+	return generations, nil
+}
+
+func isRelevantGenerationEvent(event shared.CliEvent) bool {
+	if event.SourceRevisionDigest == nil {
```

Comment:
In the future we should pull down the `workflow.yaml` and inspect it. If it's a remote URL we can re-use that. Did you try looking into `event.Workflow...`?

Reply:
I thought that the registry would be the combined / merged spec?